ledger/block/src/transactions/merkle.rs

```diff
@@ -16,9 +16,9 @@ use super::*;
 
 impl<N: Network> Transactions<N> {
     /// Returns the finalize root of the transactions.
-    pub fn to_finalize_root(&self) -> Result<Field<N>> {
+    pub fn to_finalize_root(&self, ratify_finalize_operations: Vec<FinalizeOperation<N>>) -> Result<Field<N>> {
```

Comment:
```suggestion
    pub fn to_finalize_root(&self, ratify_finalize_operations: &[FinalizeOperation<N>]) -> Result<Field<N>> {
```
that, or even `I: Iterator<Item = &FinalizeOperation<N>>`